crates/steel-core/src/primitives/ports.rs

```diff
@@ -29,10 +33,21 @@ pub fn port_module() -> BuiltInModule {
         .register_native_fn_definition(IS_INPUT_DEFINITION)
         .register_native_fn_definition(IS_OUTPUT_DEFINITION)
         .register_native_fn_definition(DEFAULT_INPUT_PORT_DEFINITION)
-        .register_native_fn_definition(DEFAULT_OUTPUT_PORT_DEFINITION);
+        .register_native_fn_definition(DEFAULT_OUTPUT_PORT_DEFINITION)
+        .register_native_fn_definition(CLOSE_OUTPUT_PORT_DEFINITION)
+        .register_native_fn_definition(DEFAULT_ERROR_PORT_DEFINITION)
+        .register_native_fn_definition(EOF_OBJECT_DEFINITION)
+        .register_native_fn_definition(OPEN_INPUT_STRING_DEFINITION)
+        .register_native_fn_definition(READ_BYTE_DEFINITION)
+        .register_native_fn_definition(READ_CHAR_DEFINITION)
+        .register_native_fn_definition(WRITE_BYTE_DEFINITION)
+        .register_native_fn_definition(WRITE_BYTES_DEFINITION)
+        .register_native_fn_definition(PEEK_BYTE_DEFINITION);
     module
 }
 
+// TODO: implement textual-port? and binary-port?
+
```

Comment:
Were you planning on doing this in this PR?

Reply:
No, I took the liberty to leave a couple of TODOs.

There are a few things that I believe would require a deeper refactor of ports, so I stopped before this PR got too unwieldy. FWIW, I was thinking about:

* Textual/binary split
* `peek-char`
* Bidirectional ports (that are both input and output and can be closed independently).

Reply:
Got it - is the implementation then just fundamentally incompatible with `peek-char`? I haven't looked at ports in a bit so I'd have to refresh my memory

Reply:
(I also remembered another one: `char-ready?`, `u8-ready?`)

re: `peek-char`: I actually added some context [here](https://github.com/mattwparas/steel/pull/227/files#diff-0102212dc5c4ef8c26b71d2b0b861ad2ef745eb95081fed3b29b54f95f68bb6bR270-R272).

TBF I'm not sure how fundamental the changes would need to be, I just went for quick wins and avoided more convoluted work for this PR.